pkg/apis/cilium/v1alpha1/types_network.go

```diff
@@ -219,6 +219,15 @@ type NetworkConfig struct {
 	// SnatOutOfCluster enables the masquerading of packets outside of the cluster
 	// +optional
 	SnatOutOfCluster *SnatOutOfCluster `json:"snatOutOfCluster,omitempty"`
+	// EnableIPv4Masquerade masquerades packets from endpoints leaving the host with BPF instead of iptables if Snat is not enabled
+	// +optional
+	EnableIPv4Masquerade *bool `json:"enableIpv4Masquerade,omitempty"`
```

Comment:
I'm a bit confused about this option. Is it actually used anywhere?
At least in he config map it just seems to depend on `.Values.global.ipv4.enabled` https://github.com/gardener/gardener-extension-networking-cilium/blob/5d1b91448c248ef352a005edfaa658be3e966941/charts/internal/cilium/charts/config/templates/configmap.yaml#L334-L338.